test/Unit/App/Middleware/DeprecationMiddlewareTest.php

```diff
@@ -185,6 +206,196 @@ public function process(
+        };
+
+        $routeResult           = $this->createMock(RouteResult::class);
+        $route                 = $this->createMock(Route::class);
+        $lazyLoadingMiddleware = new LazyLoadingMiddleware(
+            $this->createMock(MiddlewareContainer::class),
+            $handler::class,
+        );
+
+        $route->method('getMiddleware')->willReturn($lazyLoadingMiddleware);
+        $routeResult->method('isFailure')->willReturn(false);
+        $routeResult->method('getMatchedRoute')->willReturn($route);
+        $this->request->method('getAttribute')->with(RouteResult::class)->willReturn($routeResult);
+        $this->request->method('getMethod')->willReturn(RequestMethodInterface::METHOD_GET);
+        $this->handler->method('handle')->with($this->request)->willReturn($this->response);
+
+        $response = $this->subject->process($this->request, $this->handler);
+
+        $this->assertTrue($response->hasHeader('link'));
+        $this->assertFalse($response->hasHeader('sunset'));
+
+        $this->assertSame($this->formatLink(self::VERSIONING_CONFIG['application']['versioning']['documentation_url'], [
+            'rel'  => 'sunset',
+            'type' => 'text/html',
+        ]), $response->getHeader('link')[0]);
+    }
+
+    /**
+     * @throws ReflectionException
+     * @throws Exception
+     */
+    public function testDynamicLink(): void
+    {
+        $handler = new #[ResourceDeprecation(link : 'dynamic-link')] class implements RequestHandlerInterface {
+            public function handle(ServerRequestInterface $request): ResponseInterface
+            {
+                return new EmptyResponse();
+            }
+        };
+
+        $routeResult           = $this->createMock(RouteResult::class);
+        $route                 = $this->createMock(Route::class);
+        $lazyLoadingMiddleware = new LazyLoadingMiddleware(
+            $this->createMock(MiddlewareContainer::class),
+            $handler::class,
+        );
+
+        $route->method('getMiddleware')->willReturn($lazyLoadingMiddleware);
+        $routeResult->method('isFailure')->willReturn(false);
+        $routeResult->method('getMatchedRoute')->willReturn($route);
+        $this->request->method('getAttribute')->with(RouteResult::class)->willReturn($routeResult);
+        $this->request->method('getMethod')->willReturn(RequestMethodInterface::METHOD_GET);
+        $this->handler->method('handle')->with($this->request)->willReturn($this->response);
+
+        $response = $this->subject->process($this->request, $this->handler);
+
+        $this->assertTrue($response->hasHeader('link'));
+        $this->assertFalse($response->hasHeader('sunset'));
+
+        $this->assertSame($this->formatLink('dynamic-link', [
+            'rel'  => 'sunset',
+            'type' => 'text/html',
+        ]), $response->getHeader('link')[0]);
+    }
+
+    /**
+     * @throws ReflectionException
+     * @throws Exception
+     */
+    public function testSunset()
```

Comment:
```suggestion
    public function testSunset(): void
```
psalm: MissingReturnType: Method ApiTest\Unit\App\Middleware\DeprecationMiddlewareTest::testSunset does not have a return type, expecting void